src/transformStateWithClones.js

```diff
@@ -6,8 +6,40 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
   // write code here
+  const result = [];
+  let copy = { ...state };
+
+  for (const object of actions) {
+    switch (object.type) {
+      case 'addProperties':
+        copy = Object.assign({}, copy, object.extraData);
+        break;
+
+      case 'removeProperties':
+        copy = removeProperties(copy, object.keysToRemove);
```

Comment:
The `removeProperties` function correctly creates a new object, so this line is fine. However, ensure that `copy` is reassigned to a new object each time to maintain immutability.